### ISSUE TITLE
when using a marshmallow schema as return type hint, both data and errors are returned

### ISSUE THREAD

**Issue by djoek** [timothycrosley/hug#627]
when using:

```python

@hug.get()
def items(_id: hug.types.text) -> item_schema:
    return item_db.find_one(_id)
```

the resulting json is 

```json

{"data": {...}, "errors": {}}
```

Can this be made analogous to using a marshmallow schema on input ie. if valid, data gets returned, otherwise, error?

**Comment by danigosa** [comment]
Have you tried to set the parameter `strict=True` in your schema definition that it will actually throw an exception that you can handle with the decorator `@exception` and re-raise for instance a `400 Bad Request` or whatever?

**Comment by djoek** [comment]
@danigosa that's indeed an option, but it's also for valid returned data. it's not even validated, regardless whether you return a correct schema or not, it will just wrap your data in a dict with a data and an error key

**Comment by dangitall** [comment]
This seems to not be an issue with the Marshmallow 3.0 betas, fwiw.

**Comment by djoek** [comment]
hi @dangitall , just tried it with the 3.0.0b1 of marshmallow, same issue.   I'll upload a gist

**Comment by djoek** [comment]
https://gist.github.com/djoek/575a04f855f79a14c9d2fdfcfe410b7f

**Comment by djoek** [comment]
hold on, seems i used b1 instead of b10.  That is slightly better, except that when you return an incorrect object, (required text missing) I just get an empty data structure back, instead of an errors {}

**Comment by dangitall** [comment]
@djoek I agree the behavior seems wrong, but instead of getting an errors field returned, wouldn't you expect hug to raise an exception like it does for other return type annotations?

**Comment by djoek** [comment]
@dangitall yes, exactly that, i'd like it if it's handled exactly like input types (return data on success, raise exception on errors) but as i understand, hug handles those exceptions gracefully and allows to return the error to the caller?

**Comment by dangitall** [comment]
That's not what I've seen, but I may have been doing something wrong. But it also doesn't make much sense to me to return the error to the caller. Trying to return invalid data is an error on the server programing side. It seems unhelpful an risky (from a security perspective) to be returning the server's errors to the client.

**Comment by djoek** [comment]
good point. i'll try out your patches, looks good
